include/gmsh.h

```diff
@@ -21,34 +23,46 @@ class GMSH : public Mesh {
   //! Read elements in GMSH
   void read_elements(const std::string& filename);
 
+  //! Store element id and vertices coordinates as map
+  void store_element_vertices();
+
+  //! Compute material points from element coordinate map
+  void compute_material_points();
+
   //! Compute stresses
   void compute_stresses();
 
   //! call total number of vertices generated
   unsigned nvertices() const { return nvertices_; }
 
-  //! Return a vector of vertices
-  std::vector<std::shared_ptr<Point<3>>> vertices() const { return vertices_; }
+  //! Return a map of mesh element vertices
+  std::map<unsigned , std::array<unsigned, Tdim>> vertices() const {
+    return vertices_;
+  }
 
-  //! Return a vector of elements
-  std::vector<std::shared_ptr<Point<3>>> elements() const { return elements_; }
+  //! Return a map of element id & vertices id
+  std::map<unsigned, std::array<unsigned, Tvertices>> elements() const {
+    return elements_;
+  }
 
   //! Return a vector of stresses
-  std::vector<std::array<double, 6>> stress() const { return stress_; }
+  std::vector<std::array<double, Tdim * 2>> stress() const { return stress_; }
 
  private:
-  //! Number of vertices
+  //! Number of vertices in total
   unsigned nvertices_;
 
-  //! Vector of vertices 
-  std::vector<std::shared_ptr<Point<3>>> vertices_;
+  //! Stress vector in Voigt Notation
+  //! $\sigma_{xx}$ $\sigma_{yy}$ $\sigma_{zz}$ $\tau_{yz}$ $\tau_{zx}$
+  //! $\tau_{xy}$
+  std::vector<std::array<double, Tdim * 2>> stress_;
 
-  //! Vector of elements
-  std::vector<std::shared_ptr<Point<3>>> elements_;
+  //! Map to store id and vertices coordinates
+  std::map<double, std::array<double, Tdim>> vertices_;
 
-  //! Stress vector in Voigt Notation
-  //! $\sigma_{xx}$ $\sigma_{yy}$ $\sigma_{zz}$ $\tau_{yz}$ $\tau_{zx}$ $\tau_{xy}$
-  std::vector<std::array<double, 6>> stress_;
+  //! Map to store element ID and vertices ID
+  std::map<double, std::array<double, Tvertices>> elements_;
+  std::map<double, std::array<double, Tdim * Tvertices>> elementcoordinates_;
```

Comment:
Why coordinates of elements, is it not sufficient to map it to the ID of vertices?